evidence/reactor_test.go

```diff
@@ -470,96 +475,33 @@ func TestReactorBroadcastEvidence_FullyConnected(t *testing.T) {
 		stateDBs[i] = initializeValidatorState(t, val, height)
 	}
 
-	testSuites := createTestSuites(t, stateDBs, 0)
+	rts := setup(t, stateDBs, 0)
+	rts.start(t)
 
-	// Simulate a router by listening for all outbound envelopes and proxying the
-	// envelopes to the respective peer (suite).
-	wg := new(sync.WaitGroup)
-	for _, suite := range testSuites {
-		simulateRouter(wg, suite, testSuites, numEvidence*(len(testSuites)-1))
-	}
-
-	evList := createEvidenceList(t, testSuites[0].pool, val, numEvidence)
+	evList := createEvidenceList(t, rts.pools[rts.network.RandomNode().NodeID], val, numEvidence)
 
 	// every suite (reactor) connects to every other suite (reactor)
-	for _, suiteI := range testSuites {
-		for _, suiteJ := range testSuites {
-			if suiteI.peerID != suiteJ.peerID {
-				suiteI.peerUpdatesCh <- p2p.PeerUpdate{
+	for outerID, outerChan := range rts.peerChans {
+		for innerID := range rts.peerChans {
+			if outerID != innerID {
+				outerChan <- p2p.PeerUpdate{
 					Status: p2p.PeerStatusUp,
-					NodeID: suiteJ.peerID,
+					NodeID: innerID,
 				}
 			}
 		}
 	}
 
 	// wait till all suites (reactors) received all evidence from other suites (reactors)
-	waitForEvidence(t, evList, testSuites...)
+	rts.waitForEvidence(t, evList)
 
-	for _, suite := range testSuites {
-		require.Equal(t, numEvidence, int(suite.pool.Size()))
+	for _, pool := range rts.pools {
+		require.Equal(t, numEvidence, int(pool.Size()))
 
 		// commit state so we do not continue to repeat gossiping the same evidence
-		state := suite.pool.State()
+		state := pool.State()
 		state.LastBlockHeight++
-		suite.pool.Update(state, evList)
-	}
-
-	wg.Wait()
-}
-
-func TestReactorBroadcastEvidence_RemovePeer(t *testing.T) {
```

Comment:
Why did we remove this test?

Reply:
I talked this through with @alexanderbez, but it seemed to us that this was a test of the router's behavior when a reactor disconnects and then reconnects. Not only is this maybe a bit suspect, but there's no real way, without exposing a lot of internals of the router, to replicate the behavior under test.